Add newInterpolator

kleisauke opened the pull request:
Usage:
```php
require __DIR__ . '/vendor/autoload.php';
use Jcupitt\Vips;

$image = Vips\Image::newFromFile($argv[1]);
$interp = Vips\Image::newInterpolator('bicubic');
$imgAffine = $image->affine([2, 0, 0, 2], ['interpolate' => $interp]);

$widthInput = $image->width;
$widthOutput = $imgAffine->width;
if ($widthInput * 2 == $widthOutput) {
    echo 'pass';
}
$image->writeToFile($argv[2]);
```

jcupitt commented:
That's great! Could you add a test as well? There should be a reference to this somewhere in the main docs too.

kleisauke commented:
I've added some documentation for resample (derived from [here](https://jcupitt.github.io/libvips/API/current/libvips-resample.html#libvips-resample.description)) and for the `thumbnail` operation. Does this looks OK?

Also added a test case (which fails because version 1.0.7 of the extension has not been released yet). Will do a `git squash` when 1.0.7 is out (then it triggers a Travis CI rebuild).

jcupitt commented:
It all looks good to me! I'll do a pecl update and we can merge.

jcupitt commented:
OK, pecl update done, I'll merge, it should pass.
